code/modules/mob/living/carbon/xenomorph/xenomorph.dm

```diff
@@ -411,3 +408,45 @@
 	SIGNAL_HANDLER
 	var/obj/alien/weeds/found_weed = locate(/obj/alien/weeds) in loc
 	loc_weeds_type = found_weed?.type
+
+#define BURROW_FIRE_RESIST_MODIFIER 20
+
+/// Burrow code for xenomorphs
+/mob/living/carbon/xenomorph/proc/xeno_burrow()
+	SIGNAL_HANDLER
+	if(!HAS_TRAIT(src, TRAIT_BURROWED))
+		to_chat(src, span_xenowarning("We start burrowing into the ground"))
+		INVOKE_ASYNC(src, .proc/xeno_burrow_doafter, src)
+		return
+	if(HAS_TRAIT(src, TRAIT_BURROWED))
+		UnregisterSignal(src, COMSIG_MOVABLE_MOVED)
+		fire_resist_modifier += BURROW_FIRE_RESIST_MODIFIER
+		icon_state = initial(icon_state)
+		mouse_opacity = initial(mouse_opacity)
+		density = TRUE
+		throwpass = FALSE
+		canmove = TRUE
+		REMOVE_TRAIT(src, TRAIT_MOB_ICON_UPDATE_BLOCKED, src)
+		REMOVE_TRAIT(src, TRAIT_BURROWED, src)
+		REMOVE_TRAIT(src, TRAIT_HANDS_BLOCKED, src)
+
+/// Called by xeno_burrow only when burrowing
+/mob/living/carbon/xenomorph/proc/xeno_burrow_doafter()
+	if(!do_after(src, 3 SECONDS, TRUE, null, BUSY_ICON_DANGER))
+		return
+	to_chat(src, span_xenowarning("We have burrowed ourselves, we are hidden from the enemy"))
+	// This part here actually burrows the xeno
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	density = FALSE
+	throwpass = TRUE
+	icon_state = "[xeno_caste.caste_name] Burrowed" // We set it here so we dont wait for life
+	wound_overlay.icon_state = "" // We set it here so we dont wait for life
+	fire_overlay.icon_state = "" // We set it here so we dont wait for life
```

Comment:
Copy paste comment that doesnt explain anything

Wait for life? why would oyu need to wait for life?